bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -12,27 +12,10 @@
 #include <unistd.h>
 #include <utils/migrate.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-bool bsg_event_write(bsg_report_header *header, bugsnag_event *event, int fd);
-
-bugsnag_event *bsg_event_read(int fd);
-bsg_report_header *bsg_report_header_read(int fd);
-bugsnag_event *bsg_map_v4_to_report(bugsnag_report_v4 *report_v4);
-bugsnag_event *bsg_map_v3_to_report(bugsnag_report_v3 *report_v3);
-bugsnag_event *bsg_map_v2_to_report(bugsnag_report_v2 *report_v2);
-bugsnag_event *bsg_map_v1_to_report(bugsnag_report_v1 *report_v1);
-
-void migrate_app_v1(bugsnag_report_v2 *report_v2, bugsnag_report_v3 *event);
-void migrate_app_v2(bugsnag_report_v4 *report_v4, bugsnag_event *event);
-void migrate_device_v1(bugsnag_report_v2 *report_v2, bugsnag_report_v3 *event);
-void migrate_breadcrumb_v1(bugsnag_report_v2 *report_v2,
-                           bugsnag_report_v3 *event);
-
-#ifdef __cplusplus
-}
-#endif
+static bool event_write(bsg_report_header *header, bugsnag_event *event,
```

Comment:
As a general rule all symbols use a `bsg_` prefix to avoid name clashes - we've been burnt by this in the past.

Reply:
These are static functions, so they're by definition safe from symbol name clashes.